packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs

```diff
@@ -407,14 +405,17 @@ private ScmMethodProvider BuildProtocolMethod(MethodProvider createRequestMethod
             {
                 // If there are optional parameters, but the request options parameter is not optional, make the optional parameters nullable required.
                 // This is to ensure that the request options parameter is always the last parameter.
-                foreach (var parameter in optionalParameters)
+
+                for (var i = 0; i < optionalParameters.Count; i++)
                 {
-                    parameter.DefaultValue = null;
-                    parameter.Type = parameter.Type.WithNullable(true);
+                    // Ensure a new copy is made to avoid modifying the original reference.
+                    var optionalParam = optionalParameters[i].ToPublicInputParameter();
```

Comment:
Is this necessary? I would think it is okay to modify the original reference as the changes should apply to each of the protocol method overloads.